ResearchKit/Onboarding/ORKRegistrationStep.m

```diff
@@ -0,0 +1,258 @@
+        item.optional = NO;
+        
+        [formItems addObject:item];
+    }
+    
+    if (_options & ORKRegistrationStepIncludeGender) {
+        NSArray *textChoices = @[[ORKTextChoice choiceWithText:ORKLocalizedString(@"GENDER_MALE", nil) value:@0],
+                                 [ORKTextChoice choiceWithText:ORKLocalizedString(@"GENDER_FEMALE", nil) value:@1],
+                                 [ORKTextChoice choiceWithText:ORKLocalizedString(@"GENDER_OTHER", nil) value:@2]];
+        ORKValuePickerAnswerFormat *answerFormat = [ORKAnswerFormat valuePickerAnswerFormatWithTextChoices:textChoices];
+        
+        ORKFormItem *item = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemGender
+                                                               text:ORKLocalizedString(@"GENDER_FORM_ITEM_TITLE", nil)
+                                                       answerFormat:answerFormat];
+        item.placeholder = ORKLocalizedString(@"GENDER_FORM_ITEM_PLACEHOLDER", nil);
+        item.optional = NO;
+       
+        [formItems addObject:item];
+    }
+    
+    if (_options & ORKRegistrationStepIncludeDOB) {
+        // Calculate default date (20 years from now).
+        NSDateComponents *minusTwentyYears = [NSDateComponents new];
+        minusTwentyYears.year = -20;
+        NSDate *defaultDate = [[NSCalendar currentCalendar] dateByAddingComponents:minusTwentyYears
+                                                                                toDate:[NSDate date]
+                                                                               options:0];
+        
+        ORKDateAnswerFormat *answerFormat = [ORKAnswerFormat dateAnswerFormatWithDefaultDate:defaultDate
+                                                                                 minimumDate:nil
+                                                                                 maximumDate:nil
+                                                                                    calendar:nil];
+        
+        ORKFormItem *item = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemDOB
+                                                               text:ORKLocalizedString(@"DOB_FORM_ITEM_TITLE", nil)
+                                                       answerFormat:answerFormat];
+        item.placeholder = ORKLocalizedString(@"DOB_FORM_ITEM_PLACEHOLDER", nil);
+        item.optional = NO;
+        
+        [formItems addObject:item];
+    }
+    
+    return formItems;
+}
+
+- (ORKTextAnswerFormat *)passwordAnswerFormat {
+    ORKFormItem *passwordFormItem = ORKFindInArrayByFormItemId(self.formItems, ORKRegistrationFormItemPassword);
+    ORKTextAnswerFormat *passwordAnswerFormat = (ORKTextAnswerFormat *)passwordFormItem.answerFormat;
+    return passwordAnswerFormat;
+}
+
+- (NSString *)passcodeValidationRegex {
+    return [self passwordAnswerFormat].regex;
```

Comment:
Not from your code, rename `regex` to `validationRegex` on `ORKTextAnswerFormat` too? Just feel `validationRegex` is more descriptive.